### PR TITLE
Add comment to warn against early exits from freeze

### PR DESCRIPTION
**Pull request by inducer**

### CODE REVIEW

File: arraycontext/impl/pytato/__init__.py [hunk 1]
```diff
@@ -296,6 +296,10 @@ def _record_leaf_ary_in_dict(key: Tuple[Any, ...],
                     raise TypeError(f"{type(self).__name__}.freeze invoked "
                                     f"with non-pytato array of type '{type(array)}'")
 
+                # Don't be tempted to take shortcuts here for, e.g. empty
```
Comment (kaushikcfd) [inline arraycontext/impl/pytato/__init__.py:299]:
```suggestion
                # Don't be tempted to take shortcuts here, for e.g. empty
```

File: arraycontext/impl/pytato/__init__.py [hunk 2]
```diff
@@ -296,6 +296,10 @@ def _record_leaf_ary_in_dict(key: Tuple[Any, ...],
...
+                # arrays, as this will inhibit metadata propagation that
+                # may happen in transform_dag, below. See
```
Comment (kaushikcfd) [inline arraycontext/impl/pytato/__init__.py:301]:
```suggestion
                # may happen in transform_dag below. See
```
?

**Review by kaushikcfd — APPROVED** [review]
Thanks! LGTM.

File: arraycontext/impl/pytato/__init__.py (hunk 2 above)
Comment (inducer) [reply arraycontext/impl/pytato/__init__.py]: e245f3b7ded208d84eac636fbc1c1b204a94b4ca

File: arraycontext/impl/pytato/__init__.py (hunk 1 above)
Comment (inducer) [reply arraycontext/impl/pytato/__init__.py]: e245f3b7ded208d84eac636fbc1c1b204a94b4ca

**Comment by inducer** [comment]
Thanks for taking a look!

**Pull request merged**